src/testapp/sessions/universe.cpp

```diff
@@ -531,4 +533,402 @@ Session setup_testplanets_draw(
+    // rotations.
+    // TODO: Alignment is needed for SIMD (not yet implemented). see Corrade alignedAlloc
+
+    std::size_t bytesUsed = 0;
+
+    // Positions and velocities are arranged as XXXX... YYYY... ZZZZ...
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[0]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[1]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[2]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[0]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[1]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[2]);
+
+    // Rotations use XYZWXYZWXYZWXYZW...
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satRotations[0],
+        rMainSpaceCommon.m_satRotations[1],
+        rMainSpaceCommon.m_satRotations[2],
+        rMainSpaceCommon.m_satRotations[3]);
+
+    partition(bytesUsed, c_planetCount, rCoordNBody[mainSpace].mass);
+    partition(bytesUsed, c_planetCount, rCoordNBody[mainSpace].radius);
+    partition(bytesUsed, c_planetCount, rCoordNBody[mainSpace].color);
+
+    // Allocate data for all planets
+    rMainSpaceCommon.m_data = Array<unsigned char>{ Corrade::NoInit, bytesUsed };
+
+    std::size_t nextBody = 0;
+    auto const add_body = [&rMainSpaceCommon, &nextBody, &rCoordNBody, &mainSpace]
+    (
+        Vector3l position,
+        Vector3d velocity,
+        Magnum::Vector4d rotation,
+        float mass,
+        float radius,
+        Magnum::Color3 color
+        )
+        {
+            auto const [x, y, z] = sat_views(rMainSpaceCommon.m_satPositions, rMainSpaceCommon.m_data, c_planetCount);
+            auto const [vx, vy, vz] = sat_views(rMainSpaceCommon.m_satVelocities, rMainSpaceCommon.m_data, c_planetCount);
+            auto const [qx, qy, qz, qw] = sat_views(rMainSpaceCommon.m_satRotations, rMainSpaceCommon.m_data, c_planetCount);
+
+            auto const massView = rCoordNBody[mainSpace].mass.view(arrayView(rMainSpaceCommon.m_data), c_planetCount);
+            auto const radiusView = rCoordNBody[mainSpace].radius.view(arrayView(rMainSpaceCommon.m_data), c_planetCount);
+            auto const colorView = rCoordNBody[mainSpace].color.view(arrayView(rMainSpaceCommon.m_data), c_planetCount);
+
+            x[nextBody] = position.x();
+            y[nextBody] = position.y();
+            z[nextBody] = position.z();
+
+            vx[nextBody] = velocity.x();
+            vy[nextBody] = velocity.y();
+            vz[nextBody] = velocity.z();
+
+            qx[nextBody] = rotation.x();
+            qy[nextBody] = rotation.y();
+            qz[nextBody] = rotation.z();
+            qw[nextBody] = rotation.w();
+
+            massView[nextBody] = mass;
+            radiusView[nextBody] = radius;
+            colorView[nextBody] = color;
+
+            ++nextBody;
+        };
+
+    add_body // Sun
+    (
+        Vector3l{ 0, 0, 0 },
```

Comment:
minor: since these are function parameters, C++ can assume what type these are. It's possible to just use `{ 0, 0, 0 }`. may be worth combining the lines here too.